nexus/tests/integration_tests/basic.rs

```diff
@@ -242,16 +242,33 @@ async fn test_projects_basic(cptestctx: &ControlPlaneTestContext) {
     assert_eq!(project.identity.description, expected.identity.description);
     assert!(project.identity.description.len() > 0);
 
-    // Delete "simproject2".  We'll make sure that's reflected in the other
-    // requests.
+    // Delete "simproject2", but first delete:
+    // - The default subnet within the default VPC
+    // - The default VPC
+    NexusRequest::object_delete(
```

Comment:
Out of curiosity, do we have (and/or does this PR add) tests for the sorts of races that `rcgen` is meant to prevent?

Reply:
Good question, and honestly, not *really*?

We do have a test on the general "rcgen should increment" behavior: https://github.com/oxidecomputer/omicron/blob/5fe955d946ed2f962f56a5278e8939c1d62ef688/nexus/src/db/collection_insert.rs#L631-L639

But we don't have a reliable way to:
- Start the delete operation
- Check that "no children exist"
- (Separate thread of execution) Insert a new child resource
- Show that deletion fails because of the rcgen mismatch

(Should we? Probably!)

Reply:
This is where tokio's [`time::pause`](https://docs.rs/tokio/latest/tokio/time/fn.pause.html) can come in handy -- you can use it to insert arbitrary interleavings between tasks. This can also be used with property-based tests -- we should maybe consider this at some point in the future :)